### PR TITLE
feat: new fee structure

### PR DESCRIPTION
**Pull request by ielashi**
Updates the fee structure to make pricing more fair and more future-proof, per the recent benchmarking work.

### CODE REVIEW

File: canister/src/types.rs [hunk 1]
```diff
@@ -54,10 +54,34 @@ impl Default for Config {
 
 #[derive(CandidType, Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Default)]
 pub struct Fees {
-    pub get_utxos: u128,
+    /// The base fee to charge for all `get_utxos` requests.
+    pub get_utxos_base: u128,
```
Comment (ielashi) [inline canister/src/types.rs:58]: Note that these changes will require us to temporarily add some `serde[default]`s. I'll be adding those, along with hard-coding the actual fees we'll use for the launch, in a separate commit.

Comment (dsarlis) [reply canister/src/types.rs:58]: What about the `get_utxos` field you're removing? Isn't that a potential issue with candid?

File: canister/src/lib.rs [hunk 2]
```diff
@@ -160,6 +160,18 @@ pub(crate) fn genesis_block(network: Network) -> Block {
...
+
+    let amount: u64 = amount.try_into().expect("amount must be u64");
+    assert_eq!(
+        msg_cycles_accept(amount),
+        amount,
+        "Accepting cycles must succeed"
+    );
+}
+
+/// Panics if the request less than the amount of cycles given.
```
Comment (dsarlis) [inline canister/src/lib.rs:173]:
```suggestion
/// Panics if the request contains less than the amount of cycles given.
```

**Review by dsarlis — APPROVED** [review]

File: canister/src/types.rs (hunk 1 above)
Comment (ielashi) [reply canister/src/types.rs:58]: It does break the `set_config` API, but that's a private endpoint so that isn't an issue.

Comment (dsarlis) [reply canister/src/types.rs:58]: I'm also not entirely sure that decoding an old state with the new struct would work. You tested specifically the upgrade of the bitcoin canister to this version on the testnet?

**Pull request merged**